File: stickers-java/src/main/java/me/delev/stickers/StickerController.java
package me.delev.stickers;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/stickers")
public class StickerController {

    private final StickerService service;

    public StickerController(StickerService service) {
        this.service = service;
    }

    @PostMapping
    public Sticker create(@RequestBody String name) {
        return service.create(name);
    }

    @GetMapping
    public List<Sticker> list() {
        return service.findAll();
    }

    @GetMapping("/{id}")
    public ResponseEntity<Sticker> find(@PathVariable String id) {
        return service.findById(id)
                .map(ResponseEntity::ok)
                .orElseGet(() -> ResponseEntity.notFound().build());
    }
}

File: stickers-kotlin/src/main/kotlin/me/delev/stickers/StickerController.kt
package me.delev.stickers

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/stickers")
class StickerController(val service: StickerService) {

    @GetMapping
    fun getAll(): List<Sticker> {
        return service.findAll()
    }

    @PostMapping
    fun create(@RequestBody title: String): Sticker {
        return service.create(title)
    }

    @GetMapping("/{id}")
    fun getById(@PathVariable id: String): ResponseEntity<Sticker> {
        return service.findById(id)
                .map { ResponseEntity.ok(it) }
                .orElseGet { ResponseEntity.notFound().build() }
    }
}